### ISSUE TITLE
memory leak

### ISSUE THREAD

**Issue by jaxjaxk** [Zet0rz/nzombies#328]
we have a serious memory leak issue.
about every 24 hours i get an error "error error  not enough memory". i run garbage collection every round so something is staying that should not be staying.

**Comment by jaxjaxk** [comment]
also this is folowed by a lua panic. might want to look over this
https://facepunch.com/showthread.php?t=1486766

**Comment by Zet0rz** [comment]
Never encountered this before, and never got it reported either. You sure it isn't caused by one of your modifications? Then again, I've never run a server for 24 hours, which you btw should not do. Regardless, if you can provide more info it would greatly help. It could very well be linked with #314

**Comment by jaxjaxk** [comment]
@Zet0rz it would not be fair to people if if had to stop the server every 24 hours to reboot. it is not as if the server does not already to a soft reboot every map change.
i would love to believe it is related to #314 however that error generates a different message and crashes clients 9/10 times without crashing the server.

what might help is that i use a server host service. should the server fail to respond enough times the server reboots and i am notified. both times this happened i was not notified and the server never restarted. it is as if the server was responsive to pings but could do nothing else.

**Comment by minin43** [comment]
No, it's an internal issue within the Source engine, a server cannot be run for longer than ~6 hours with issues arising. All you have to do is map change every so once in awhile.

**Comment by Zet0rz** [comment]
> it is not as if the server does not already to a soft reboot every map change.

How often do you change map?

**Comment by sammyt291** [comment]
We restart all our build servers each 24hrs to avoid massive performance/stability drops. have not had any need to restart more often though in the last two years of hosting 4 - 7 servers :)

**Comment by func-octo** [comment]
> we have a serious memory leak issue.
about every 24 hours i get an error "error error not enough memory". i run garbage collection every round so something is staying that should not be staying.

Well, the best way to keep a server from crashing is to restart it every once in awhile, like @sammyt291 said. I would restart the server every 4 hours, as the time in between restarts is ample for play, and it only takes a few minutes for a server to restart.
